src/receiver.rs

```diff
@@ -31,7 +25,6 @@ impl<R> Receiver<R> {
 }
 
 enum Inner<R> {
-    Disconnected,
     Receiving(MapErr<catty::Receiver<R>, fn(catty::Disconnected) -> Disconnected>),
     Done,
 }
```

Comment:
If this has only two states now, is it even worth having this enum?

I think we can hold the `MapErr` as a regular field now and just poll that.

Reply:
Together with that, I think this code should move into `send_future`, esp. now that we have multiple `Receiver`s flying around :D

Reply:
> I think we can hold the MapErr as a regular field now and just poll that.

I think that returns Poll::Pending after completed instead of panicking - we can move away from this if that's acceptable

Reply:
> > I think we can hold the MapErr as a regular field now and just poll that.
> 
> I think that returns Poll::Pending after completed instead of panicking - we can move away from the inner enum if that's acceptable

I think that is fine. No one should expect any specific behaviour from futures after their completion.

Reply:
Done now